src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -520,15 +551,23 @@ private OOBPayload parseOOB(String oobPayload) {
         ServiceInterface s = Runtime.getService(payload.getServiceName());
         if (s == null) {
           log.warn("Service name in OOB/MRL tag unknown. {}", mrlPayload);
+
+          if (payload.getParams() != null) {
+            conversationHistory.add("<font color='#FF0000'><b> -> Missing OOB : " + payload.getParams() + "</b></font>");

```

Comment:
HTML markup should not be programmatically specified here. conversation history should be fixed length queue of response objects.